osf/models/node.py

```diff
@@ -1351,17 +1354,16 @@ def add_contributor_registered_or_not(self, auth, user_id=None,
                                  permissions=permissions, send_email=send_email, save=True)
         else:
             contributor = get_user(email=email)
-            if contributor:
-                if self.contributor_set.filter(user=contributor).exists():
-                    raise ValidationValueError('{} is already a contributor.'.format(contributor.fullname))
-                self.add_contributor(contributor=contributor, auth=auth, visible=bibliographic,
-                                    send_email=send_email, permissions=permissions, save=True)
+            if contributor and self.contributor_set.filter(user=contributor).exists():
+                raise ValidationValueError('{} is already a contributor.'.format(contributor.fullname))
+
+            if contributor and contributor.is_registered:
+                    self.add_contributor(contributor=contributor, auth=auth, visible=bibliographic,
+                                        send_email=send_email, permissions=permissions, save=True)
             else:
-                contributor = self.add_unregistered_contributor(
-                    fullname=full_name, email=email, auth=auth,
-                    send_email=send_email, permissions=permissions,
-                    visible=bibliographic, save=True
-                )
+                contributor = self.add_unregistered_contributor(fullname=full_name, email=email, auth=auth,
```

Comment:
nit pick, but we don't really need to change these lines just for style reasons. I only mention just to minimize lines changed for a hotfix I guess.